daffodil-core/src/main/scala/edu/illinois/ncsa/daffodil/dsom/Term.scala

```diff
@@ -48,13 +46,17 @@ import edu.illinois.ncsa.daffodil.schema.annotation.props.gen.NilKind
 /////////////////////////////////////////////////////////////////
 
 // A term is content of a group
```

Comment:
scaladoc. Term, and what is a term and what isn't a term, is key to all of DSOM.

ElementRef and LocalElementDecl are the two concrete kinds of element Term. GlobalElementDecl is not a Term. 

This was done so that property resolving and statement aggregation can only be done on Term, which then forces one to only obtain properties from Term.

Similarly for groups...

SequenceGroupRef and Sequence are Term, GlobalSequenceGroupDef is not. Similarly for choices.
Perhaps add dsom package doc explaining this?